packages/client-generator-ts/src/TSClient/PrismaClient.ts

```diff
@@ -292,17 +292,18 @@ export type LogOptions<ClientOptions extends Prisma.PrismaClientOptions> =
 export interface PrismaClientConstructor {
   ${indent(this.jsDoc, TAB_SIZE)}
   new <
-    ClientOptions extends Prisma.PrismaClientOptions = Prisma.PrismaClientOptions,
-    const U = LogOptions<ClientOptions>,
+    Options extends Prisma.PrismaClientOptions,
+    LogOpts extends LogOptions<Options>,
+    OmitOpts extends Prisma.PrismaClientOptions['omit'] = Options extends { omit: infer U } ? U : Prisma.PrismaClientOptions['omit'],
     ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs
-  >(options?: Prisma.Subset<ClientOptions, Prisma.PrismaClientOptions>): PrismaClient<ClientOptions, U, ExtArgs>
+  >(options?: Options): PrismaClient<LogOpts, OmitOpts, ExtArgs>
```

Comment:
`Prisma.Subset` was checking that no extra unknown properties were provided. Is removing it intentional? I don’t feel strongly about it, but want to make sure it was a conscious decision that was made for a reason.